lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -953,6 +959,39 @@ ChangeResult TypeAnalyzer::visitAtenMaxPool2dOp(
   return getLatticeElement(op->getResult(0)).join(knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitAtenMaxPool2dWithIndicesOp(
+    AtenMaxPool2dWithIndicesOp op,
+    ArrayRef<LatticeElement<ValueKnowledge> *> operands) {
+  auto self = operands[0]->getValue();
+
+  // Contains Knowledge of shape and dtype for the 1st result.
+  auto outputKnowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op.getContext());
+  outputKnowledge.hasSizes = true;
+  auto &ifm = operands[0]->getValue();
+  SmallVector<int64_t, 2> kernelSize;
+  if (!matchPattern(op.kernel_size(), m_TorchConstantIntList(kernelSize)))
+    kernelSize = SmallVector<int64_t, 2>{kUnknownSize, kUnknownSize};
+  if (ifm.hasSizes)
+    outputKnowledge.sizes = computeOpWithKernelOutputShape(
+        op, ifm, ifm.sizes[1], kernelSize[0], kernelSize[1]);
+  else
+    outputKnowledge.sizes.resize(4, kUnknownSize);
+  outputKnowledge.dtype = operands[0]->getValue().dtype;
```

Comment:
This seems to be shared by both `AtenMaxPool2d` and `AtenMaxPool2dWithIndices`, so it should be refactored.